pkg/docker/config/config.go

```diff
@@ -201,13 +201,13 @@ func getAuthFilePaths(sys *types.SystemContext, homeDir string) []authPath {
 			authPath{path: filepath.Join(dockerConfig, "config.json"), legacyFormat: false},
 		)
 	} else {
+		paths = append(paths,
+			authPath{path: filepath.Join(homeDir, dockerLegacyHomePath), legacyFormat: true},
+		)
```

Comment:
So is it the last path entry 'wins', right (i.e. the latter overwrites a former entry if both are present in the filesystem). Then the change looks good to me.

Reply:
I actually can not see where last entry wins.  BTW, on a quick look.  It looks to me the first one with a username/password would win.  Although I am not an expert on this code.

Reply:
Yes, the first entry wins. Which makes this PR hard to reconcile with the original bug report.

Reply:
```go
// getAuthFilePaths returns a slice of authPaths based on the system context
// in the order they should be searched. 
```
and `TestGetAuthPreferNewConfig`.

Reply:
Switched to use legacy last.

Reply:
Sorry that my issue description is so vague.  Unfortunately, I don't have a reproducer anymore, but my problem was solved after I have removed the legacy `~/.dockercfg` that I still had parallel to `~/.docker/config.json`. I will restore it from my backup and will add a more in-depth description of the issue.